.github/workflows/ci.yml

```diff
@@ -7,9 +7,30 @@ on:
   pull_request:
     branches:
       - '*'
+permissions:
+  contents: read
 
 jobs:
+  filter:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    outputs:
+      run_tests: ${{ steps.filter.outputs.run_tests }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3.0.2
+        id: filter
+        with:
+          predicate-quantifier: 'every'
+          filters: |
+            run_tests:
+              - '!**/*.md'
+              - '!tools/**'
+              - '!perfmetrics/**'
+              - '!samples/**'
   format-test:
+    needs: filter
+    if: ${{ needs.filter.outputs.run_tests == 'true' }}
```

Comment:
Formatting is cheap. I'd also like integration tests and other golang code to be correctly formatted when they are changed. So, let's remove this filtering from here.

Same comment for lint as well.